File: python/day1/main.py
import itertools
import functools
import re
import math

from copy import deepcopy

INPUT_FILE = 'input.txt'

def read_file(filename, func=None):
    result = []
    with open(filename, 'r') as f:
        for line in f:
            result.append(func(line.strip()) if func else line.strip())
    return result

def main1():
    lines = read_file(INPUT_FILE, int)
    increased = 0
    for idx in range(1, len(lines)):
        if lines[idx] > lines[idx - 1]:
            increased += 1
    print(increased)

def main2():
    lines = read_file(INPUT_FILE, int)
    increased = 0
    for idx in range(3, len(lines)):
        if sum(lines[idx-3:idx]) < sum(lines[idx-2:idx+1]):
            increased += 1
    print(increased)
    
if __name__ == "__main__":
    main1()
    main2()

File: rust/day1/src/main.rs
use day1::util;

fn main() {
    let lines = util::read_ints("input.txt");
    part1(&lines);
    part2(&lines);
}

fn part1(lines: &Vec<i32>) {
    let mut increased = 0;
    for pos in 1..lines.len() {
        if lines[pos - 1] < lines[pos] {
            increased += 1;
        }
    }
    println!("{}", increased);
}


fn part2(lines: &Vec<i32>) {
    let mut increased = 0;
    for pos in 3..lines.len() {
        let a:i32 = lines[pos-3..pos].iter().sum();
        let b:i32 = lines[pos-2..=pos].iter().sum();
        if a < b {
            increased += 1;
        }
    }
    println!("{}", increased);
}
